src/components/UserFilterDropdown/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {useEffect, useState, ChangeEvent} from 'react';
+import {useDispatch} from 'react-redux';
+import {displayErrorToast} from 'utils/toasts';
+import {AppDispatch, SFC} from 'types';
+import * as S from './Styles';
+import axios from 'axios';
+import {UserReadSerializer} from 'types';
+import {authorizationHeaders} from 'utils/authentication';
+const BASE_URL = `${process.env.REACT_APP_API_URL}/api/users`;
+
+interface User {
+  id: number;
+  username: string;
+  avatar: string;
+}
+const UserFilterDropdown: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [filteredUsers, setFilteredUsers] = useState<User[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const dispatch = useDispatch<AppDispatch>();
+
+  useEffect(() => {
+    // Filter users based on the search term
+    const filterUsers = () => {
+      const filtered = users.filter((user) => user.username.toLowerCase().includes(searchTerm.toLowerCase()));
+      setFilteredUsers(filtered);
+    };
+
+    filterUsers();
+  }, [searchTerm, users]);
+
+  const handleSearch = async (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value);
+    if (event.target.value == '') {
+      setUsers([]);
+      setFilteredUsers([]);
+      return;
+    }
+
+    try {
+      const response = await axios.get<UserReadSerializer, any>(
+        `${BASE_URL}?search=${event.target.value}`,
+        authorizationHeaders(),
+      );
```

Comment:
I don't think it's a good idea to hit the API directly from the component.
We should use a dispatcher instead